onnxruntime/core/providers/webgpu/buffer_manager.cc

```diff
@@ -297,11 +319,31 @@ WGPUBuffer BufferManager::Create(size_t size, wgpu::BufferUsage usage) {
   return buffer;
 }
 
+WGPUBuffer BufferManager::CreateUMA(size_t size, wgpu::BufferUsage usage) {
```

Comment:
Can we embed this into BufferManager.create()? If we need to createUMA() for other purposes in the future, it's easier to extend. Now we need to duplicate the conditions (!session_initialized_ && context_.SupportsBufferMapExtendedUsages()) here and there.
In the near future, I think we need to calculate all the memory consumption clearly (tracking their creation and release), to have a single place to manage the buffer would make it easier.

Reply:
Embedding this into `BufferManager.create()` could not help reduce the conditions unless we can move them to `BufferManager`. However `session_initialized` shouldn't be inside `BufferManager` as discussed earlier.

For the near future consideration, I don't have strong preference on this. it's up to you guys. :)

Reply:
Then new `BufferManager.create()` would have 2 new parameters like this:
```
  WGPUBuffer Create(size_t size,
                    wgpu::BufferUsage usage = wgpu::BufferUsage::Storage | wgpu::BufferUsage::CopySrc |
                                              wgpu::BufferUsage::CopyDst,
                    bool mapped_at_creation = false, bool acquire_cached = false);
```
Let me if you guys prefer this way!

Reply:
I'm fine with current implementation given BufferManager is not aware of stages.